[Page Add][Enhanced Function] SenseCAP Indicator applies the Matter 1.0 protocol

# This assignment has been successfully achieved 🎉

Thanks Tim and the wiki page is: https://wiki.seeedstudio.com/SenseCAP_Indicator_Application_Matter/

## Overview
- Use [SenseCAP Indicator](https://www.seeedstudio.com/SenseCAP-Indicator-D1-p-5643.html) to apply the Matter 1.0 protocol and connect with iPhone or Google services(HomeKit).
- Use iPhone and control the local sensors with Matter.
- Indicator's onboard sensors, Grove sensors, and proprietary protocol LoRa node devices are converted to Matter-enabled devices
## Assignment acceptance
- Please check the "assignee" section before accepting to see if this has already been accepted.
- Please ensure that the **requested timeline** aligns with your workflow.
- Leave your **Github Name** below to indicate that you are willing to accept the job.
- If you are taking on the assignment **for the first time**, you may receive an invitation in your GitHub mail.
## Requirements(check list)
- Prove of video presentation
- Detailed operation procedure written in **Markdown file** with [wiki standards](https://files.seeedstudio.com/wiki/wiki-ranger/Standards_of_Product_ApplicationWiki_Upload.pdf)
- Upload wiki markdown file to GitHub repository with [guide & instructions](https://wiki.seeedstudio.com/Contribution-Guide/)
- List of products used
- Code files that complete the program
## What do we offer?
We want to express our gratitude to our contributors by offering a range of rewards.  Depending on the difficulties(Tier 0/1/2/3), the time we expect(Days 1/3/7/15),  the actual  submission time and the actual submission contents, we will offer different levels of rewards.
1. We will provide our products or cash bonus to our contributors as a token of appreciation.
2. For those contributors who join our "Ranger Program"(announced soon), we offer even more exciting rewards, including access to our prototype products.
3. In addition, all contributors and their contributions will be acknowledged and showcased on our wiki platform.
## FAQ
For the ranger who accepts this assignment can have a conversation with your transition AE below

Hello, Timo @Timo614  , Next I will test your program.  

When I compiled your project via (esp-idf: v5.0.1  esp-matter: v1.1 ), an error was reported. May I ask if there are other things that need to be paid attention to in the compilation environment? 

```
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Users/virgil/.espressif/tools/xtensa-esp32s3-elf/esp-2022r1-11.2.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32s3
Solving dependencies requirements
CMake Error at /Users/virgil/seeed/esp-idf/tools/cmake/build.cmake:519 (message):
  ERROR: HTTP request error

Call Stack (most recent call first):
  /Users/virgil/seeed/esp-idf/tools/cmake/project.cmake:440 (idf_build_process)
  CMakeLists.txt:29 (project)
```

--- a/docs/Sensor/SenseCAP/SenseCAP_Indicator/Application/Matter.md
+++ b/docs/Sensor/SenseCAP/SenseCAP_Indicator/Application/Matter.md
@@ -1,0 +1,153 @@
+---
+description: SenseCAP Indicator x Matter Application Development
+title: Matter X SenseCAP Indicator
+keywords:
+- SenseCAP Indicator
+- Matter
+- ESP32S3
+image: https://files.seeedstudio.com/wiki/wiki-platform/S-tempor.png
+slug: /SenseCAP_Indicator_Application_Matter
+last_update:
+  date: 8/28/2023
+  author: Tim
+---
+
+# SenseCAP Indicator x Matter Application Development
+<!-- SenseCAP Indicator x Matter Application Development -->
+
+<iframe class="youtube-video" src="https://www.youtube.com/embed/LCIWqwmCZ54" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+
+## Introduction
+
+This is a demo of how to use the [Matter SDK](https://project-chip.github.io/connectedhomeip-doc/index.html) to build enriched applications for your SenseCAP Indicator.
+
+The SenseCAP Indicator has a touchscreen which makes it capable of acting as a powerful home automation interface. Matter makes the process easier than ever as it takes away a lot of the complicated aspects of configuration. With the help of Matter's SDK one can quickly scan a QR code and configure their device onto their home network.
+
+For this article we'll go through the steps involved to get your device up and running with the demo.
+
+<br />
+
+<div class="get_one_now_container" style={{textAlign: 'center'}}>
+    <a class="get_one_now_item" href="https://www.seeedstudio.com/SenseCAP-Indicator-D1-p-5643.html">
+            <strong><span><font color={'FFFFFF'} size={"4"}> Get One Now 🖱️</font></span></strong>
+    </a>
+</div>
+
+<br />
+
+The Matter SDK provides a standard set of devices and allows access to their data and the uploading of their state to the controller. ESP-Matter was developed by Espressif as an official Matter development framework for ESP32 series SoCs.
+
+To get your SenseCAP Indicator working with Matter, you need to follow a few main steps:
+
+1. [Install esp-idf and esp-matter](#install_sdks)
+2. [Configuring your environment and building](#configure_and_build)
+3. [Using the application](#using_application)
+4. [Configuring the Home Assistant UI further](#going_beyond)
+
+## Prerequisites
+
+Before we begin, make sure you have read the [User Guide](/Sensor/SenseCAP/SenseCAP_Indicator/Get_started_with_SenseCAP_Indicator) of the SenseCAP Indicator Board to familiarize yourself with its software and hardware information.
+
+## Install ESP IDF and ESP Matter {#install_sdks}
+
+You'll want to follow the [instructions provided by the ESP-Matter SDK documentation](https://docs.espressif.com/projects/esp-matter/en/latest/esp32/developing.html#getting-the-repositories) and install version 5.0.1 of ESP-IDF and the release version 1.1 from ESP-Matter.
+
+After those version has been installed please install the patch provided by Seeed Studio [apply the provided patch](https://github.com/Seeed-Solution/SenseCAP_Indicator_ESP32/tree/main/tools/patch).
+
+## Configuring your environment and building {#configure_and_build}
+
+### Using the CLI to configure and build
+
+```sh
+~/esp/esp-idf/install.sh
+~/.espressif/esp-matter/install.sh
+
+. ~/esp/esp-idf/export.sh
+. ~/.espressif/esp-matter/export.sh
+
+cd examples/indicator_matter
+idf.py set-target esp32s3
+idf.py fullclean
+idf.py erase-flash
+idf.py build flash monitor
+```
+
+## Using the application {#using_application}
+
+Once the firmware has been flashed the screen starts with just the QR code. Open your home automation's phone application and enter into the device provisioning flow (varies depending upon manufacturer).
+
+In the following example you can see my use of the Home Assistant mobile application to connect to my Home Assistant Yellow running the Matter Beta service. 
+
+<div align="center"><img width={480} src="https://files.seeedstudio.com/wiki/SenseCAP/SenseCAP_Indicator/indicator-matter-1.JPG"/></div>
+
+Upon scanning the device my phone immediately begins the process of provisioning. The provisioning process handled securely passing your credentials to the device allowing it to connect to the home automation controller device over wifi. In this way you never need to actually write the wifi credentials anywhere.
+
+After provisioning the device will begin reporting its temperature and humidity to the home assistant controller at intervals of 10 seconds.
+
+### Standard functionality
+
+<iframe class="youtube-video" src="https://www.youtube.com/embed/mBhrYeyQQeg" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+
+The Indicator Matter application allows for browsing sensor data in a similar way to the standard firmware. Upon loading the device shows the clock screen which is configured based upon your location to give you an accurate time. The screen has events to detect being slid to the right and left to cycle through the time, sensor display, and settings screen.
+
+
+<div class="table-center">
+  <table align="center">
+    <tr>
+        <td><div style={{textAlign:'center'}}><img src="https://files.seeedstudio.com/wiki/SenseCAP/SenseCAP_Indicator/indicator-matter-2.JPG" style={{width:480, height:'auto'}}/></div></td>
+        <td><div style={{textAlign:'center'}}><img src="https://files.seeedstudio.com/wiki/SenseCAP/SenseCAP_Indicator/indicator-matter-3.JPG" style={{width:480, height:'auto'}}/></div></td>
+    </tr>
+    <tr>
+        <td><div style={{textAlign:'center'}}><img src="https://files.seeedstudio.com/wiki/SenseCAP/SenseCAP_Indicator/indicator-matter-4.JPG" style={{width:480, height:'auto'}}/></div></td>
+        <td><div style={{textAlign:'center'}}><img src="https://files.seeedstudio.com/wiki/SenseCAP/SenseCAP_Indicator/indicator-matter-5.JPG" style={{width:480, height:'auto'}}/></div></td>
+    </tr>
+  </table>
+</div>
+
+Clicking a sensor further shows additional information about the state of that sensor and the history can be viewed in this way. 
+
+The settings screen allows configuration of several elements of the device's function. A 12 or 24 hour clock can be configured, the timezone is overridable, and the display brightness is adjustable.
+
+### Matter Home Assistant Dashboard
+
+To access the Matter Home Assistant Dashboard swipe down from the main clock screen. This dashboard has multiple different components which are associated with demo devices on the Matter side. During the provisioning process these devices have endpoints created which can be used for accessing common features such as dimming a light or turning it off completely.
+
+<div align="center"><img width={480} src="https://files.seeedstudio.com/wiki/SenseCAP/SenseCAP_Indicator/indicator-matter-6.JPG"/></div>
+
+In the case of this demo two dimmable lights were setup along with a door lock. You can see as you click the buttons the home assistant side is also updated to reflect these changes. The sliders can be moved to set the brightness of the light.
+
+From the home assistant side the devices can be enabled and disabled or modified as needed and the SenseCAP Indicator also is updated to reflect it. In this way you can have additional automations disable or enable functionality on your Indicator device and have the screen reflect their state in realtime.
+
+The virtual dashboard and automation capabilities of the home controller in use allow for various interesting combinations. With the SenseCAP Indicator you can build dashboards of your own imagination to control devices across your home environment.
+
+<div align="center"><img width={480} src="https://files.seeedstudio.com/wiki/SenseCAP/SenseCAP_Indicator/indicator-matter-7.JPG"/></div>
+
+## Configuring the Home Assistant UI further {#going_beyond}
+
+The UI itself is built using [LVGL](https://lvgl.io/). For more detailed information about configuring LVGL see the [following article](https://wiki.seeedstudio.com/using_lvgl_and_tft_on_round_display/).
+
+The UI for the dashboard is composed of LGVL components with events setup for callbacks such that their data is persisted both to local storage and to the matter endpoint. The virtual dashboard controller (`indicator_virtual_dashboard_controller.c`) handles updating the form and posting associated events for other consumers when the underlying data is changed.
+
+For the initial example three devices have been setup and the underlying logic in the matter model handles responding to events from the view and persisting the data on to the matter endpoints or updating the state of the UI.
+
+In general the following considerations must be made when updating the UI and persisting to Matter:
+- During the initialization process the method `indicator_matter_setup()` of the `indicator_matter.cpp` model sets up any associated devices for use with Matter. In addition a callback is setup for view events such that the matter endpoints are updated on change.
+- The `indicator_virtual_dashboard.c` model listens to view events associated with the virtual dashboard and persists the data to storage for retreival on initialization.
+- The matter logic subscribes to the events around modified matter data (whether remotely or locally)
+- The virtual dashboard controller handles updating the state of the form as needed in response to events sent via the `indicator_matter.cpp` Matter callbacks on changes from Matter controller.
+
+## Resources
+
+1. **Demo SDK**: The Demo SDK for the SenseCAP Indicator is available on [GitHub](https://github.com/Seeed-Solution/SenseCAP_Indicator_ESP32).
+2. **User Guide**: The User Guide provides detailed information about the software and hardware of the SenseCAP Indicator Board. You can read it [here](/Sensor/SenseCAP/SenseCAP_Indicator/Get_started_with_SenseCAP_Indicator).
+3. **Getting Started Guide for ESP-IDF**: This guide provides full steps to configure and use ESP-IDF to build projects. You can access it [here](https://docs.espressif.com/projects/esp-idf/en/latest/get-started/index.html).
+4. **Matter Installation Guide**: If you're new to ESP-Matter, this guide will help you get it installed and set up. You can find it [here](https://docs.espressif.com/projects/esp-matter/en/latest/esp32/developing.html).
+
+
+## Tech Support
+
+**Need help with your SenseCAP Indicator? We're here to assist you!**
+
+If you encounter any issues or have any questions while following this tutorial, please feel free to reach out to our tech support. We are always here to help!
+
+Visit our [Seeed Official Discord Channel](https://discord.gg/sensecap) to ask your questions or the [GitHub discussions](https://github.com/Seeed-Solution/SenseCAP_Indicator_ESP32/discussions) to share all you want!

--- a/sidebars.js
+++ b/sidebars.js
@@ -1367,6 +1367,7 @@ const sidebars = {
             'Sensor/SenseCAP/SenseCAP_Indicator/Application/OpenAI',
             'Sensor/SenseCAP/SenseCAP_Indicator/Application/Home_Assistant',
             'Sensor/SenseCAP/SenseCAP_Indicator/Application/LoRa',
+            'Sensor/SenseCAP/SenseCAP_Indicator/Application/Matter',
           ],
         },
       ],

